src/Client.php

```diff
@@ -51,12 +57,30 @@ public function __construct(string $accessToken, GuzzleClient $client = null, in
     {
         $this->accessToken = $accessToken;
 
-        $this->client = $client ?? new GuzzleClient();
+        $this->client = $client ?? new GuzzleClient(['handler' => self::createHandler()]);
 
         $this->maxChunkSize = ($maxChunkSize < self::MAX_CHUNK_SIZE ? ($maxChunkSize > 1 ? $maxChunkSize : 1) : self::MAX_CHUNK_SIZE);
         $this->maxUploadChunkRetries = $maxUploadChunkRetries;
     }
 
+    /**
+     * Create a new guzzle handler stack.
+     *
+     * @return \GuzzleHttp\HandlerStack
+     */
+    protected static function createHandler()
+    {
+        $stack = HandlerStack::create();
+
+        $stack->push(Middleware::retry(function ($retries, $request, $response = null, $exception = null) {
+            return $retries < 3 && ($exception instanceof ConnectException || ($response && ($response->getStatusCode() >= 500 || in_array($response->getStatusCode(), self::RETRY_CODES, true))));
+        }, function ($retries) {
+            return (int) pow(2, $retries) * self::RETRY_BACKOFF);
```

Comment:
Missing grouping bracket:
```php
(int) pow(...
```
should be
```php
(int) (pow(...
```

Reply:
The bracket is not missing. It's not needed. Only the output of pow needs casting.

Reply:
Oh, there's not meant to be a bracket there at all actually. ;)